sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -2296,7 +2297,20 @@ case class GpuFormatNumber(x: Expression, d: Expression)
     val d = rhs.getValue.asInstanceOf[Int]
     x.dataType match {
       case FloatType | DoubleType => {
-        CastStrings.fromFloatWithFormat(lhs.getBase, d)
+        val nanSymbol = DecimalFormatSymbols.getInstance(Locale.US).getNaN
+        // JDK 8's nan symbol is "�" ('\uFFFD'), which is also the jni kernel's nan symbol
+        // In higher JDK version, nan symbol is "NaN", so we need to handle it here.
```

Comment:
We already have a custom kernel to handle formatting, and it's not hard to pass parameters to CUDA kernels.  Is there a plan to fix this directly in the kernel at some point?  Needing to do fixups after the custom kernel increases memory pressure and reduces performance, and we have the ability to update the custom kernel do the right thing from the start.

Reply:
Yes, fix in the kernel is better, I plan to do it soon in a follow-up if you think ok. I chose a quicker fix in this PR to unblock JDK 11 & 17 nightly CI.

Reply:
Yes that's totally fine, just checking that this isn't considered the final fix.  Thanks!